src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,108 @@
 /* eslint-disable */
 import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import axios from 'axios';
+const reciver = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ZEAeqeXuXcX9bQfaMnAJ/books'
 
+export const getBookAsync = createAsyncThunk('book/gethBooks', async () => {
+  try {
+    const response = await axios.get(reciver);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
 
-const initialState = [
-  {
-    "item_id": "item1",
-    "title": "The Great Gatsby",
-    "author": "John Smith",
-    "category": "Fiction"
-  },
-  {
-    "item_id": "item2",
-    "title": "Anna Karenina",
-    "author": "Leo Tolstoy",
-    "category": "Fiction"
-  },
-  {
-    "item_id": "item3",
-    "title": "The Selfish Gene",
-    "author": "Richard Dawkins",
-    "category": "Nonfiction"
+export const deleteBookAsync = createAsyncThunk('book/deleteBook', async (id) => {
+  try {
+    const response = await axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ZEAeqeXuXcX9bQfaMnAJ/books/${id}`,
+    {method:'DELETE'});
+    return response;
+  } catch (error) {
+    return error;
+  }
+});
+
+export const addBookAsync = createAsyncThunk('book/addBook', async (book) => {
+  try {
+    const response = await axios.post(reciver, book);
+    return response.data;
+  } catch (error) {
+    return error;
```

Comment:
- [ ] Great job so far @Yosefgeda :muscle: However, I noticed that the `addBook` and `removeBook` features don't work the same as before. I am unable to delete a book after I add a book and after refreshing the page, the book disappears and the page is empty so kindly fix these as well so your app can work smoothly :wink: 

- Video Reference

[Screencast from 04-07-2023 03:27:39 PM.webm](https://user-images.githubusercontent.com/102699958/230637406-c6d641ed-a8ca-45eb-87dc-9b1b04b43930.webm)